src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -0,0 +1,1475 @@
+        // recreate the box
+        startCutting();
+        return;
+    }
+    Part::Box* pcBox = static_cast<Part::Box*>(CutBox);
+    // get its placement and size
+    Base::Placement placement = pcBox->Placement.getValue();
+    Base::Vector3d BoxPosition = placement.getPosition();
+    // flip the box
+    if (ui->flipZ->isChecked())
+        BoxPosition.z = BoxPosition.z + pcBox->Height.getValue();
+    else
+        BoxPosition.z = BoxPosition.z - pcBox->Height.getValue();
+    placement.setPosition(BoxPosition);
+    pcBox->Placement.setValue(placement);
+
+    auto CutObject = doc->getObject(CutZName);
+    // there should be a cut, but maybe the user deleted it meanwhile
+    if (!CutObject) {
+        Base::Console().Warning((std::string("SectionCut warning: there is no ")
+            + std::string(CutZName) + std::string(", trying to recreate it\n")).c_str());
+        // recreate the box
+        startCutting();
+        return;
+    }
+    Part::Cut* pcCut = static_cast<Part::Cut*>(CutObject);
+    pcCut->recomputeFeature(true);
+}
+
+// refreshes the list of document objects and the visible objects
+void SectionCut::onRefreshCutPBclicked()
+{
+    // get document
+    doc = Gui::Application::Instance->activeDocument()->getDocument();
+    if (!doc) {
+        Base::Console().Error("SectionCut error: there is no document\n");
+        return;
+    }
+    // get all objects in the document
+    std::vector<App::DocumentObject*> ObjectsList = doc->getObjects();
+    if (ObjectsList.empty()) {
+        Base::Console().Error("SectionCut error: there are no objects in the document\n");
+        return;
+    }
+    // empty the ObjectsListVisible
+    ObjectsListVisible.clear();
+    // now store those that are currently visible
+    for (auto it = ObjectsList.begin(); it != ObjectsList.end(); ++it) {
+        if ((*it)->Visibility.getValue())
+            ObjectsListVisible.push_back((*it));
+    }
+    // reset defaults
+    hasBoxX = false;
+    hasBoxY = false;
+    hasBoxZ = false;
+    // we can have existing cuts
+    if (doc->getObject(CutZName)) {
+        hasBoxZ = true;
+        blockSignals(true);
+        ui->groupBoxZ->setChecked(true);
+        blockSignals(false);
+    }
+    if (doc->getObject(CutYName)) {
+        hasBoxY = true;
+        blockSignals(true);
+        ui->groupBoxY->setChecked(true);
+        blockSignals(false);
+    }
+    if (doc->getObject(CutXName)) {
+        hasBoxX = true;
+        blockSignals(true);
+        ui->groupBoxX->setChecked(true);
+        blockSignals(false);
+    }
+    // if there is a cut, disable the button
+    if (hasBoxX || hasBoxY || hasBoxZ)
+        ui->RefreshCutPB->setEnabled(false);
+}
+
+SbBox3f SectionCut::getViewBoundingBox()
+{
+    auto docGui = Gui::Application::Instance->activeDocument();
+    Gui::View3DInventor* view = static_cast<Gui::View3DInventor*>(docGui->getActiveView());

```

Comment:
@donovaly The static_cast is unsafe because the active view can be something else, e.g. a spreadsheet view. Use a dynamic_cast and check the result.